src/interfaces/IPlugin.sol

```diff
@@ -109,17 +112,18 @@ interface IPlugin {
     /// account.
     function onUninstall(bytes calldata data) external;
 
-    /// @notice Run the pre user operation validation hook specified by the `functionId`.
-    /// @dev Pre user operation validation hooks MUST NOT return an authorizer value other than 0 or 1.
-    /// @param functionId An identifier that routes the call to different internal implementations, should there be
-    /// more than one.
-    /// @param userOp The user operation.
-    /// @param userOpHash The user operation hash.
-    /// @return Packed validation data for validAfter (6 bytes), validUntil (6 bytes), and authorizer (20 bytes).
-    function preUserOpValidationHook(uint8 functionId, PackedUserOperation calldata userOp, bytes32 userOpHash)
-        external
-        returns (uint256);
+    /// @notice Describe the contents and intended configuration of the plugin.
+    /// @dev This manifest MUST stay constant over time.
+    /// @return A manifest describing the contents and intended configuration of the plugin.
+    function pluginManifest() external pure returns (PluginManifest memory);
+
+    /// @notice Describe the metadata of the plugin.
+    /// @dev This metadata MUST stay constant over time.
+    /// @return A metadata struct describing the plugin.
+    function pluginMetadata() external pure returns (PluginMetadata memory);
+}
 
+interface IValidation is IPlugin {
```

Comment:
Thoughts on suffixing these interfaces with `Plugin` (e.g.,`IValidationPlugin`)? Also, if we're sticking to "validation function" as the term we want to use, might be clearer to call this `IValidationFunction(Plugin)`.

Reply:
Eh maybe leaving the suffix out is closer to the interfaces exposed by 7579. Gets us closer to alignment there.

Reply:
Yeah, we could use the "validator" term instead, but I remember objections from the Safe team due to it being the same word as a validator in proof-of-stake.

Reply:
I'm fine with how we have it now!